src/kitovu/sync/filecache.py

```diff
@@ -0,0 +1,70 @@
+"""FileCache keeps track of the state of the files, remotely and locally. It exists to determine if the local file
+has been changed between two synchronisation processes and allows for a conflict handling accordingly."""
+
+from enum import Enum
+import json
+import pathlib
+import typing
+
+from kitovu.sync import syncplugin
+from kitovu import utils
+
+# sync-process:
+# every time we sync something, the FileCache gets read, upon which kitovu decides what needs to be downloaded.
+# The files get downloaded, for each file, the FileCache gets updated for the files that were synchronised.
+
+
+class Filestate(Enum):
+    """Used to discern in which places files have changed"""
+    LOCAL = 1
+    REMOTE = 2
+    BOTH = 3
+    NONE = 4
+
+
+class File:
+    def __init__(self, local_digest_at_synctime: str,
+                 relative_path_with_filename: pathlib.PurePath,
+                 plugin: syncplugin.AbstractSyncPlugin) -> None:
+        file_properties = {"path": relative_path_with_filename,
+                           "plugin": plugin,
+                           "digests": {
+                                "cached digest": local_digest_at_synctime,
+                                }
+                           }
+
+
+class FileCache:
+    FILECACHE: str = "filecache.json"
```

Comment:
It'd be good if the filename (and path) could be passed to the constructor (`__init__`) instead. That'll make it easier to test as you can use a `FileCache` with a temporary file easily.